subprojects/core/src/main/java/org/gradle/api/internal/initialization/DefaultScriptHandler.java

```diff
@@ -158,6 +159,12 @@ private void defineConfiguration() {
         }
         if (classpathConfiguration == null) {
             classpathConfiguration = configContainer.migratingUnlocked(CLASSPATH_CONFIGURATION, ConfigurationRolesForMigration.LEGACY_TO_RESOLVABLE_DEPENDENCY_SCOPE);
+            configContainer.beforeCollectionChanges(methodName ->
```

Comment:
🤔 If we have the methodName available here, why not work that into the error message itself?  This would help people who encounter this error quickly figure out what's going on with it.

Reply:
I probably should be able to integrate this into the message